CreditManagement/models.py

```diff
@@ -100,73 +99,417 @@ def __str__(self):
         return "{} | {} | {} → {} | {}".format(self.moment, self.amount, self.source(), self.target(), self.notes)
 
 
-# Todo: this is not yet in use
-class AbstractCreditQuerySet(models.QuerySet):
-    source_column = None
-    target_column = None
+class UserWithCredit(User):
+    """
+    User model enhanced with credit queries.
+    """
+    class Meta:
+        proxy = True
+
+
+
+    def get_balance(self):
+        return -119
+
+
+class AssociationWithCredit(Association):
+    """
+    Association model enhanced with credit queries.
+    """
+    class Meta:
+        proxy = True
+
+    def get_balance(self):
+        return -121
+
+
+"""""""""""""""""""""""""""""""""""""""""""""
+New implementation of the transaction models
+"""""""""""""""""""""""""""""""""""""""""""""
+
+
+class AbstractTransaction(models.Model):
+    """
+    Abstract model defining the Transaction models, can retrieve information from all its children
+    """
+    # DO NOT CHANGE THIS ORDER, IT CAN CAUSE PROBLEMS IN THE UNION METHODS AT DATABASE LEVEL!
+    source_user = models.ForeignKey(User, related_name="%(class)s_transaction_source", on_delete=models.SET_NULL, null=True, blank=True, verbose_name="The user giving the money")
+    source_association = models.ForeignKey(Association, related_name="%(class)s_transaction_source", on_delete=models.SET_NULL, null=True, blank=True, verbose_name="The association giving the money")
+    amount = models.DecimalField(verbose_name="Money transferred", decimal_places=2, max_digits=4, validators=[MinValueValidator(Decimal('0.01'))])
+    target_user = models.ForeignKey(User, related_name="%(class)s_transaction_target", on_delete=models.SET_NULL, null=True, blank=True, verbose_name="The user receiving the money")
+    target_association = models.ForeignKey(Association, related_name="%(class)s_transaction_target", on_delete=models.SET_NULL, null=True, blank=True, verbose_name="The association recieving the money")
```

Comment:
Watch out for line length, that should be at most 120 characters. (Actually even at most 79 characters according to the Python style guide but that's kind of unworkable.)

Reply:
Fixed